app/models/pglogical_subscription.rb

```diff
@@ -86,16 +86,24 @@ def pglogical(refresh = false)
     self.class.pglogical(refresh)
   end
 
-  def self.validate(host, port, user, password, dbname, id)
+  def self.validate(host: nil, port: nil, user: nil, password: nil, dbname: nil, id: nil)
```

Comment:
Maybe do:
`def self.validate(id = nil, connection_hash)` 
Then get host, port, user, password, and dbname from the hash?

FYI a method with that signature can be called like this:
```ruby
def test_hash(id = nil, hash)
  puts id.inspect
  puts hash
end

test_hash(1, :a => "a test")
1
{:a=>"a test"}

test_hash(:a => "a test", :b => "more")
nil
{:a=>"a test", :b=>"more"}
```